### PR TITLE
config: add split-region-max-num in config to control the maximum split region number

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
* add split-region-max-num in config to control the maximum split region number.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

### CODE REVIEW

**Comment by crazycs520** [comment]
@winkyao @bb7133 PTAL

File: config/config.toml.example [hunk 1]
```diff
@@ -60,6 +60,9 @@ enable-table-lock = false
 # delay-clean-table-lock is used to control the time (Milliseconds) of delay before unlock the table in the abnormal situation.
 delay-clean-table-lock = 0
 
+# Maximum split regions numbers in split region syntax.
```
Comment (winkyao) [inline config/config.toml.example]: Maximum number of the splitting region, which is used by the split table statement

Comment (crazycs520) [reply config/config.toml.example]: done.

**Pull request merged**